src/clippoly/src/clippoly.cpp

```diff
@@ -0,0 +1,271 @@
+  }
+  return Clippoly(circle, true);
+}
+
+Clippoly Clippoly::Boolean(const Clippoly& second, OpType op) const {
+  auto ct = cliptype_of_op(op);
+  auto res = C2::BooleanOp(ct, C2::FillRule::NonZero, paths_, second.paths_,
+                           precision_);
+  return Clippoly(res, true);
+}
+
+Clippoly Clippoly::BatchBoolean(const std::vector<Clippoly>& clippolys,
+                                OpType op) {
+  if (clippolys.size() == 0)
+    return Clippoly();
+  else if (clippolys.size() == 1)
+    return clippolys[0];
+
+  auto subjs = clippolys[0].paths_;
+  auto clips = C2::PathsD();
+  for (int i = 1; i < clippolys.size(); ++i) {
+    auto ps = clippolys[i].paths_;
+    clips.insert(clips.end(), ps.begin(), ps.end());
+  }
+
+  auto ct = cliptype_of_op(op);
+  auto res = C2::BooleanOp(ct, C2::FillRule::NonZero, subjs, clips, precision_);
+  return Clippoly(res, true);
+}
+
+/**
+ * Shorthand for Boolean Union.
+ */
+Clippoly Clippoly::operator+(const Clippoly& Q) const {
+  return Boolean(Q, OpType::Add);
+}
+
+/**
+ * Shorthand for Boolean Union assignment.
+ */
+Clippoly& Clippoly::operator+=(const Clippoly& Q) {
+  *this = *this + Q;
+  return *this;
+}
+
+/**
+ * Shorthand for Boolean Difference.
+ */
+Clippoly Clippoly::operator-(const Clippoly& Q) const {
+  return Boolean(Q, OpType::Subtract);
+}
+
+/**
+ * Shorthand for Boolean Difference assignment.
+ */
+Clippoly& Clippoly::operator-=(const Clippoly& Q) {
+  *this = *this - Q;
+  return *this;
+}
+
+/**
+ * Shorthand for Boolean Intersection.
+ */
+Clippoly Clippoly::operator^(const Clippoly& Q) const {
+  return Boolean(Q, OpType::Intersect);
+}
+
+/**
+ * Shorthand for Boolean Intersection assignment.
+ */
+Clippoly& Clippoly::operator^=(const Clippoly& Q) {
+  *this = *this ^ Q;
+  return *this;
+}
+
+Clippoly Clippoly::Translate(glm::vec2 v) {
+  auto ps = C2::TranslatePaths(paths_, v.x, v.y);
+  return Clippoly(ps, true);
+}
+
+Clippoly Clippoly::Scale(glm::vec2 scale) {
+  auto scaled = C2::PathsD();
+  for (auto path : paths_) {
+    auto s = C2::PathD();
+    for (auto p : path) {
+      s.push_back(C2::PointD(p.x * scale.x, p.y * scale.y));
+    }
+    scaled.push_back(s);
+  }
+  return Clippoly(scaled, false);
+}
+
+Clippoly Clippoly::Simplify(double epsilon) {
+  auto ps = SimplifyPaths(paths_, epsilon, false);
+  return Clippoly(ps, false);
+}
+
+Clippoly Clippoly::RamerDouglasPeucker(double epsilon) {
+  auto ps = C2::RamerDouglasPeucker(paths_, epsilon);
+  return Clippoly(ps, false);
+}
+
+Clippoly Clippoly::StripNearEqual(double epsilon) {
+  auto ps = C2::StripNearEqual(paths_, epsilon, true);
+  return Clippoly(ps, false);
+}
+
+Clippoly Clippoly::StripDuplicates() {
+  auto ps = C2::StripDuplicates(paths_, true);
+  return Clippoly(ps, false);
+}
+
+Clippoly Clippoly::TrimCollinear() {
+  auto trimmed = C2::PathsD();
+  for (auto p : paths_) {
+    trimmed.push_back(C2::TrimCollinear(p, false));
```

Comment:
A bunch of these simplification methods sound very similar. Mind adding some comments describing the differences?

Reply:
RMD commented on above is an existing algorithm, and simplify is Clipper2's own take on similar. From what I recall they work at the similarity of consecutive segments (short, similar angle, can be replaced by one segment). Collinear just removes points that are fully collinear without getting into those types of calculations.

Reply:
I don't love having five functions that do basically the same thing. Let's be opinionated in our API and pick one. Since Manifold is based on inexact geometry, there's no point in exact checks anyway.

Reply:
Simplify might be enough on its own then. I think that might be designed to catch the others anyway, I'll look at the source again.